test/validation/org/apache/cassandra/bridges/Bridge.java

```diff
@@ -20,10 +20,44 @@
  */
 package org.apache.cassandra.bridges;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public abstract class Bridge
 {
+    private static final Logger logger = LoggerFactory.getLogger(Bridge.class);
+    protected final Runtime runtime = Runtime.getRuntime();
+
     public abstract void stop();
     public abstract void destroy();
     public abstract String readClusterLogs();
     public abstract void captureLogs(String testName);
+
+    public void nodeTool(int node, String command, String arguments)
+    {
+        try
+        {
+            String fullCommand = "ccm node" + node + " nodetool " + command + " " + arguments;
```

Comment:
The modules will now have access to their Bridge, that is true. But, with inheritance, that means they can run any subclass of Bridge. So yes, Bridge should have an abstract function, and this implementation should be in CCMBridge.